src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -628,6 +629,539 @@ def test_call_get_callees_local(fortran_reader):
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    arg_idx_list = []
+    result: Routine = call_foo.get_callee(ret_arg_match_list=arg_idx_list)
+
+    assert len(arg_idx_list) == 3
+    assert arg_idx_list[0] == 2
+    assert arg_idx_list[1] == 0
+    assert arg_idx_list[2] == 1
+
+    assert result is routine_match
+
+
+def test_call_get_callee_5_optional_and_named_arguments(fortran_reader):
+    '''
+    Check that optional and named arguments have been correlated correctly
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(b=e, a=f)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b
+    integer, optional :: c
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    arg_idx_list = []
+    result: Routine = call_foo.get_callee(ret_arg_match_list=arg_idx_list)
+
+    assert len(arg_idx_list) == 2
+    assert arg_idx_list[0] == 1
+    assert arg_idx_list[1] == 0
+
+    assert result is routine_match
+
+
+def test_call_get_callee_6_interfaces(fortran_reader):
+    '''
+    Check that optional and named arguments have been correlated correctly
+    '''
+    code = '''
+module some_mod
+  implicit none
+
+  interface foo
+    procedure foo_a, foo_b, foo_c
+  end interface
+contains
+
+  subroutine main()
+    integer :: e_int, f_int, g_int
+    real :: e_real, f_real, g_real
+
+    ! Should match foo_a
+    call foo(e_int, f_int)
+
+    ! Should match foo_a
+    call foo(e_int, f_int, g_int)
+
+
+    ! Should match foo_b
+    call foo(e_real, f_int)
+
+    ! Should match foo_b
+    call foo(e_real, f_int, g_int)
+
+    ! Should match foo_b
+    call foo(e_real, c=f_int, b=g_int)
+
+
+    ! Should match foo_c
+    call foo(e_int, f_real, g_int)
+
+    ! Should match foo_c
+    call foo(b=e_real, a=f_int)
+
+    ! Should match foo_c
+    call foo(b=e_real, a=f_int, g_int)
+  end subroutine
+
+  subroutine foo_a(a, b, c)
+    integer :: a, b
+    integer, optional :: c
+  end subroutine
+
+  subroutine foo_b(a, b, c)
+    real :: a
+    integer :: b
+    integer, optional :: c
+  end subroutine
+
+  subroutine foo_c(a, b, c)
+    integer :: a
+    real :: b
+    integer, optional :: c
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    if 1:
```

Comment:
Please remove all these `if`s and `print` statements (but do add comments describing what specific case each part of the test is for).

Reply:
These if statements are very handy for debugging potential future problems. Setting some of them to '0' allows you to skip this subtest.

Also, the print statements provide helpful output again for debugging.
I don't understand why there should be no output. Is this a convention from PSyclone?

Reply:
PS: The indentation with the "if 1" statements also structures such subtests nicely, hence, improving readability :-)

Reply:
Yes, not having output from tests is a convention we have. It keeps things tidy. If a test is functioning correctly then it isn't needed and if a test fails then you're going to have to do some debugging anyway.

Regarding readability, I agree that is an issue with a test as long as this but I don't think these `if`s are the way to go. A better solution is to break the test up, possibly by using `@pytest.mark.parameterize` to provide the different `call` statements one at a time. Have a grep through the various test files to see how we use this. Breaking it up into smaller chunks also helps with debugging, as you say.